src/object.hpp

```diff
@@ -32,6 +32,18 @@ namespace _impl {
     class ObjectNotifier;
 }
 
+enum class CreatePolicy : int8_t {
+    // Do not create objects if given something that could be converted to a
+    // Realm object but isn't. Used for things like find().
+    Skip,
```

Comment:
This feels a bit weird, but I guess the only other option would be packing the enum class into an `Optional` which would make it rather annoying to use.

Reply:
Agreed that it's a bit weird but I can't think of anything clearly better.